hours/importer/base.py

```diff
@@ -220,43 +228,53 @@ def save_resource(self, data: dict) -> Resource:
 
         return obj
 
-    # @db.transaction.atomic
-    # def save_period(self, data: dict) -> Period:
-    #     """
-    #     Takes the serialized Period data with Openings, creates and/or updates the
-    #     corresponding Period object, saves and returns it.
-    #     """
-    #     obj = self._update_or_create_object(Period, data)
-    #     if obj._created:
-    #         obj.save()
-    #         print("%s created" % obj)
-
-    #     # Update openings after the period has been created
-    #     openings = obj.openings.all()
-    #     openings.delete()
-    #     new_openings = []
-    #     for opening in data.get("openings", []):
-    #         # openings have no identifiers in kirkanta and they are generated from
-    #         # data, therefore, we cannot identify existing openings with new ones
-    #         new_opening = Opening(
-    #             period=obj,
-    #             weekday=opening["weekday"],
-    #             week=opening["week"],
-    #             status=opening["status"],
-    #             description=opening.get("description", None),
-    #             opens=opening.get("opens", None),
-    #             closes=opening.get("closes", None),
-    #         )
-    #         new_openings.append(new_opening)
-    #     Opening.objects.bulk_create(new_openings)
-    #     obj._changed = True
-    #     obj._changed_fields.append("openings")
-    #     if not obj._created:
-    #         self.logger.debug("%s changed: %s" %
-    #                           (obj, ", ".join(obj._changed_fields)))
-    #
-    #     # Saving updates the daily hours for the duration of the period
-    #     obj.save()
+    @db.transaction.atomic
+    def save_period(self, data: dict) -> DatePeriod:
+        """Takes the serialized period data and creates a DatePeriod
+
+        Will delete previously existing periods with the same name and dates.
+        """
+        try:
+            time_span_groups_data = data.pop("time_span_groups")
+        except KeyError:
+            time_span_groups_data = []
+
+        resource = data["resource"]
+
+        # Delete existing date period and all time spans
+        # TODO: Update existing period instead of deleting all the previous ones
+        for period in DatePeriod.all_objects.filter(
+            resource=resource,
+            name=data["name"],
```

Comment:
This logic will probably need some work in the future, e.g. if the name or any boundaries changed we would get duplicate period data?

Reply:
Yeah. I thought that we would add a DatePeriodOrigin in the future to fix that.